sprocs/course_permissions_insert_by_user_uid.sql

```diff
@@ -18,16 +18,16 @@ BEGIN
         RAISE EXCEPTION 'no user with uid: %', uid;
     END IF;
 
-    BEGIN
-        INSERT INTO course_permissions AS cp
-            (user_id, course_id, course_role)
-        VALUES
-            (user_id, course_id, course_role)
-        RETURNING
-            cp.* INTO new_row;
-    EXCEPTION
-        WHEN unique_violation THEN RAISE EXCEPTION 'user already in course, uid: %', uid;
-    END;
+    INSERT INTO course_permissions AS cp
+        (user_id, course_id, course_role)
+    VALUES
+        (user_id, course_id, course_role)
+    ON CONFLICT ON CONSTRAINT course_permissions_user_id_course_id_key
```

Comment:
Do we need the `ON CONSTRAINT ...` here? Normally you can just say `ON CONFLICT` and postgres will automatically figure our the constraint.

Reply:
`ON CONFLICT DO UPDATE` is invalid (error: `ON CONFLICT DO UPDATE requires inference specification or constraint name`). I tried `ON CONFLICT (user_id, course_id)` (which is what most other sprocs use), but that gave an error due to `user_id` being ambiguous in this context (and adding the table doesn't work, invalid syntax). One option is to use `ON CONFLICT (user_id, course_id)` and rename the local variables to something that doesn't conflict, is that better?

Reply:
Ok, found a different way to do it with column names instead.

Reply:
Nice!